cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go

```diff
@@ -51,105 +51,160 @@ func (t Template) controllersTrustPolicy() *iamv1.PolicyDocument {
+			},
+			Condition: iamv1.Conditions{
+				iamv1.StringLike: map[string]string{"iam:AWSServiceName": "elasticloadbalancing.amazonaws.com"},
+			},
+		},
+		{
+			Effect:   iamv1.EffectAllow,
+			Resource: t.allowedEC2InstanceProfiles(),
+			Action: iamv1.Actions{
+				"iam:PassRole",
+			},
+		},
+		{
+			Effect: iamv1.EffectAllow,
+			Resource: iamv1.Resources{
+				"arn:*:secretsmanager:*:*:secret:aws.cluster.x-k8s.io/*",
+			},
+			Action: iamv1.Actions{
+				"secretsmanager:CreateSecret",
+				"secretsmanager:DeleteSecret",
+				"secretsmanager:TagResource",
+			},
+		},
+	}
+	if t.Spec.ClusterAPIControllers.EKS.Enable {
+		statement = append(statement, []iamv1.StatementEntry{
 			{
-				Effect:   iamv1.EffectAllow,
-				Resource: iamv1.Resources{iamv1.Any},
 				Action: iamv1.Actions{
-					"ec2:AllocateAddress",
-					"ec2:AssociateRouteTable",
-					"ec2:AttachInternetGateway",
-					"ec2:AuthorizeSecurityGroupIngress",
-					"ec2:CreateInternetGateway",
-					"ec2:CreateNatGateway",
-					"ec2:CreateRoute",
-					"ec2:CreateRouteTable",
-					"ec2:CreateSecurityGroup",
-					"ec2:CreateSubnet",
-					"ec2:CreateTags",
-					"ec2:CreateVpc",
-					"ec2:ModifyVpcAttribute",
-					"ec2:DeleteInternetGateway",
-					"ec2:DeleteNatGateway",
-					"ec2:DeleteRouteTable",
-					"ec2:DeleteSecurityGroup",
-					"ec2:DeleteSubnet",
-					"ec2:DeleteTags",
-					"ec2:DeleteVpc",
-					"ec2:DescribeAccountAttributes",
-					"ec2:DescribeAddresses",
-					"ec2:DescribeAvailabilityZones",
-					"ec2:DescribeInstances",
-					"ec2:DescribeInternetGateways",
-					"ec2:DescribeImages",
-					"ec2:DescribeNatGateways",
-					"ec2:DescribeNetworkInterfaces",
-					"ec2:DescribeNetworkInterfaceAttribute",
-					"ec2:DescribeRouteTables",
-					"ec2:DescribeSecurityGroups",
-					"ec2:DescribeSubnets",
-					"ec2:DescribeVpcs",
-					"ec2:DescribeVpcAttribute",
-					"ec2:DescribeVolumes",
-					"ec2:DetachInternetGateway",
-					"ec2:DisassociateRouteTable",
-					"ec2:DisassociateAddress",
-					"ec2:ModifyInstanceAttribute",
-					"ec2:ModifyNetworkInterfaceAttribute",
-					"ec2:ModifySubnetAttribute",
-					"ec2:ReleaseAddress",
-					"ec2:RevokeSecurityGroupIngress",
-					"ec2:RunInstances",
-					"ec2:TerminateInstances",
-					"tag:GetResources",
-					"elasticloadbalancing:AddTags",
-					"elasticloadbalancing:CreateLoadBalancer",
-					"elasticloadbalancing:ConfigureHealthCheck",
-					"elasticloadbalancing:DeleteLoadBalancer",
-					"elasticloadbalancing:DescribeLoadBalancers",
-					"elasticloadbalancing:DescribeLoadBalancerAttributes",
-					"elasticloadbalancing:DescribeTags",
-					"elasticloadbalancing:ModifyLoadBalancerAttributes",
-					"elasticloadbalancing:RegisterInstancesWithLoadBalancer",
-					"elasticloadbalancing:DeregisterInstancesFromLoadBalancer",
-					"elasticloadbalancing:RemoveTags",
+					"iam:GetRole",
+					"iam:ListAttachedRolePolicies",
+					"iam:DetachRolePolicy",
+					"iam:DeleteRole",
+					"iam:CreateRole",
```

Comment:
I wonder if we should have an additional explicit option that the user must specify to say that are going to use the "automatic IAM role per cluster" functionality when provisioning an EKS cluster (the partner to the **EKSEnableIAM** feature flag on the controller) which would then include the actions that can modify IAM:
 - iam:DetachRolePolicy
- iam:DeleteRole
- iam:CreateRole
- iam:TagRole
- iam:AttachRolePolicy

And without that option the policy just gets:
- iam:GetRole
- iam:ListAttachedRolePolicies

Reply:
done!